optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoAnnotationWrappers.java

```diff
@@ -0,0 +1,690 @@
+                                    break;
+                                case INTEGER:
+                                    annotationValues.put(value.name(),
+                                            value.asIntArray());
+                                    break;
+                                case CHARACTER:
+                                    annotationValues.put(value.name(),
+                                            value.asCharArray());
+                                    break;
+                                case FLOAT:
+                                    annotationValues.put(value.name(),
+                                            value.asFloatArray());
+                                    break;
+                                case DOUBLE:
+                                    annotationValues.put(value.name(),
+                                            value.asDoubleArray());
+                                    break;
+                                case LONG:
+                                    annotationValues.put(value.name(),
+                                            value.asLongArray());
+                                    break;
+                                case BOOLEAN:
+                                    annotationValues.put(value.name(),
+                                            value.asBooleanArray());
+                                    break;
+                                case CLASS:
+                                    Class[] valueArr = Arrays.stream(value.asClassArray())
+                                            .map(v -> {
+                                                try {
+                                                    return Class.forName(v.toString());
+                                                } catch (ClassNotFoundException e) {
+                                                    throw new IllegalStateException(e);
+                                                }
+                                            })
+                                            .collect(Collectors.toList()).toArray(new Class[0]);
+                                    annotationValues.put(value.name(),
+                                            valueArr);
+                                    break;
+                                case STRING:
+                                    annotationValues.put(value.name(),
+                                            value.asStringArray());
+                                    break;
+                                case ENUM:
+                                    annotationValues.put(value.name(),
+                                            value.asEnumArray());
+                                    break;
+                                case ARRAY:
+                                case NESTED:
+                                case UNKNOWN:
+                                    break;
```

Comment:
Don't sweep anything under the carpet.
Either it works, or fail fast. That's the 2 options.

Reply:
Done (Nested now handled for Arrays and single; Unknown handled for arrays; Arrays of arrays and Unknown single are unsupported (and I think impossible); throws exception, along with default.